backend/compact-connect/tests/smoke/smoke_common.py

```diff
@@ -14,15 +17,153 @@ def __init__(self, message):
+    except ClientError as e:
+        logger.error(f"Failed to delete staff user data, '{email}': {str(e)}")
+        raise e
+
+
+def create_test_staff_user(*, email: str, compact: str, jurisdiction: str, permissions: dict):
+    """
+    Creates a test staff user in Cognito, stores their data in DynamoDB, and returns their user sub id.
+    """
+    logger.info(f"Creating staff user, '{email}', in {compact}/{jurisdiction}")
+    user_attributes = {'email': email, 'familyName': 'Dokes', 'givenName': 'Joe'}
+    sub = _create_staff_user_in_cognito(email=email)
+    schema = UserRecordSchema()
+    config.staff_users_dynamodb_table.put_item(
+        Item=schema.dump(
+            {
+                'type': 'user',
+                'userId': sub,
+                'compact': compact,
+                'attributes': user_attributes,
+                'permissions': permissions,
+            },
+        ),
+    )
+    logger.info(f'Created staff user record in DynamoDB. User data: {user_attributes}')
+
+    return sub
+
+
+def get_user_tokens(email, password=TEST_STAFF_USER_PASSWORD, is_staff=False):
+    """
+    Gets Cognito tokens for a user.
+    {
+        'IdToken': 'string',
+        'AccessToken': 'string',
+        'RefreshToken': 'string',
+        'ExpiresIn': 123,
+        'TokenType': 'string',
+        'NewDeviceMetadata': {
+            'DeviceKey': 'string',
+            'DeviceGroupKey': 'string'
+        }
+    }
+    """
+    try:
+        logger.info('Getting tokens for user: ' + email + ' user type: ' + ('staff' if is_staff else 'provider'))
+        response = config.cognito_client.admin_initiate_auth(
+            UserPoolId=config.cognito_staff_user_pool_id if is_staff else config.cognito_provider_user_pool_id,
+            ClientId=config.cognito_staff_user_client_id if is_staff else config.cognito_provider_user_client_id,
+            AuthFlow='ADMIN_USER_PASSWORD_AUTH',
+            AuthParameters={'USERNAME': email, 'PASSWORD': password},
+        )
+
+        return response['AuthenticationResult']
+
+    except ClientError as e:
+        logger.info(f'Failed to get tokens for user {email}: {str(e)}')
+        raise e
+
+
 def get_provider_user_auth_headers():
+    provider_token = os.environ.get('TEST_PROVIDER_USER_ID_TOKEN')
+    if not provider_token:
+        tokens = get_user_tokens(config.test_provider_user_username, config.test_provider_user_password, is_staff=False)
+        os.environ['TEST_PROVIDER_USER_ID_TOKEN'] = tokens['IdToken']
+
     return {
         'Authorization': 'Bearer ' + os.environ['TEST_PROVIDER_USER_ID_TOKEN'],
     }
 
 
-def get_staff_user_auth_headers():
+def get_staff_user_auth_headers(username: str, password: str = TEST_STAFF_USER_PASSWORD):
+    tokens = get_user_tokens(username, password, is_staff=True)
     return {
-        'Authorization': 'Bearer ' + os.environ['TEST_STAFF_USER_ACCESS_TOKEN'],
+        'Authorization': 'Bearer ' + tokens['AccessToken'],
     }
```

Comment:
Is it worth renaming one of these to make clear the distinction between them wrt caching? It seems `get_provider_user_auth_headers` only calls `get_user_tokens` if it doesn't already have a provider token, whereas `get_staff_user_auth_headers` always runs it.

e.g., `get_provider_user_auth_headers_cached`, or `get_or_fetch_provider_user_auth_headers` and `fetch_staff_user_auth_headers`